packages/client/hmi-client/src/services/data.ts

```diff
@@ -633,47 +555,37 @@ const fetchData = async (
 
 	const promiseList = [] as Promise<FullSearchResults>[];
 
-	// xdd
-	if (resultType === ResourceType.XDD || resultType === ResourceType.ALL) {
-		// eslint-disable-next-line no-async-promise-executor
-		const promise1 = new Promise<FullSearchResults>(async (resolve, reject) => {
-			try {
-				const allData = await searchXDDArticles(term, searchParam?.xdd);
-				const allDataFilteredWithFacets = await searchXDDArticles(
-					term,
-					searchParamWithFacetFilters?.xdd
-				);
-				resolve({ allData, allDataFilteredWithFacets });
-			} catch (err: any) {
-				reject(new Error(`Error fetching XDD results: ${err}`));
-			}
-		});
-		promiseList.push(promise1);
-	}
-
-	// models
-	if (resultType === ResourceType.MODEL || resultType === ResourceType.ALL) {
-		const promise2 = new Promise<FullSearchResults>((resolve, reject) => {
-			try {
-				resolve(getModels(term, searchParamWithFacetFilters?.model));
-			} catch (err: any) {
-				reject(new Error(`Error fetching model results: ${err}`));
-			}
-		});
-		promiseList.push(promise2);
-	}
-
-	// datasets
-	if (resultType === ResourceType.DATASET || resultType === ResourceType.ALL) {
-		const promise3 = new Promise<FullSearchResults>((resolve, reject) => {
-			try {
-				resolve(getDatasets(term, searchParamWithFacetFilters?.dataset));
-			} catch (err: any) {
-				reject(new Error(`Error fetching dataset results: ${err}`));
-			}
-		});
-		promiseList.push(promise3);
-	}
+	Object.entries(ResourceType).forEach(([key]) => {
+		if (resultType === ResourceType[key] || resultType === ResourceType.ALL) {
+			// eslint-disable-next-line no-async-promise-executor
+			const promise = new Promise<FullSearchResults>(async (resolve, reject) => {
+				try {
+					switch (ResourceType[key]) {
+						case ResourceType.XDD: // XDD
+							resolve({
+								allData: await searchXDDArticles(term, searchParam?.xdd),
+								allDataFilteredWithFacets: await searchXDDArticles(
+									term,
+									searchParamWithFacetFilters?.xdd
+								)
+							});
+							break;
+						case ResourceType.MODEL: // Models
+							resolve(getAssets(term, ResourceType.MODEL, searchParamWithFacetFilters?.model));
+							break;
+						case ResourceType.DATASET: // Datasets
+							resolve(getAssets(term, ResourceType.DATASET, searchParamWithFacetFilters?.dataset));
+							break;
+						default:
+							break;
+					}
+				} catch (err: any) {
+					reject(new Error(`Error fetching ${ResourceType[key]} results: ${err}`));
+				}
+			});
+			promiseList.push(promise);
+		}
+	});
```

Comment:
This will need to be moved to the `hmi-server`.
Once @Tom-Szendrey is done with his first implementation this can be a good candidate for it.